File: backend/PetCare.Modules.Booking/Domain/Booking.cs
using PetCare.Modules.IdentityAndPets.Domain;
using PetCare.Modules.Providers.Domain;

namespace PetCare.Modules.Booking.Domain;

public class Booking
{
    public Guid Id { get; set; }
    public Guid CustomerId { get; set; }
    public Customer Customer { get; set; } = null!;
    
    public Guid PetId { get; set; }
    public Pet Pet { get; set; } = null!;
    
    public Guid ProviderId { get; set; }
    public Provider Provider { get; set; } = null!;

    public string ServiceType { get; set; } = string.Empty; // e.g. Grooming, Walking, Vet
    public string Modality { get; set; } = string.Empty; // HomeVisit, PickupDropoff
    
    public DateTime ScheduledTime { get; set; }
    public string Status { get; set; } = "Pending"; // Pending, Confirmed, Completed, Cancelled
    
    public decimal TotalCost { get; set; }
}

File: frontend/lib/models/booking.dart
class Booking {
  final String id;
  final String customerId;
  final String petId;
  final String providerId;
  final String serviceType;
  final String modality;
  final DateTime scheduledTime;
  final String status;
  final double totalCost;

  Booking({
    required this.id,
    required this.customerId,
    required this.petId,
    required this.providerId,
    required this.serviceType,
    required this.modality,
    required this.scheduledTime,
    required this.status,
    required this.totalCost,
  });

  factory Booking.fromJson(Map<String, dynamic> json) {
    return Booking(
      id: json['id'] ?? '',
      customerId: json['customerId'] ?? '',
      petId: json['petId'] ?? '',
      providerId: json['providerId'] ?? '',
      serviceType: json['serviceType'] ?? '',
      modality: json['modality'] ?? '',
      scheduledTime: DateTime.parse(json['scheduledTime']),
      status: json['status'] ?? 'Pending',
      totalCost: (json['totalCost'] ?? 0).toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'customerId': customerId,
      'petId': petId,
      'providerId': providerId,
      'serviceType': serviceType,
      'modality': modality,
      'scheduledTime': scheduledTime.toIso860String(),
      'status': status,
      'totalCost': totalCost,
    };
  }
}
